app/src/main/java/org/mozilla/focus/telemetry/GleanMetricsService.kt

```diff
@@ -78,7 +79,9 @@ class GleanMetricsService(context: Context) : MetricsService {
     private fun collectPrefMetrics(
         components: Components,
         settings: Settings
-    ) = CoroutineScope(IO).async {
+    ) = CoroutineScope(Main).async {
+        // Doing this on Main because this might be the first time BrowserStore is accessed and initialized
+        // and this needs to be done on the Main thread.
```

Comment:
Hm, the comment is a bit misleading, the `Store` can be used on any thread, but the new `install()` calls can't.

The whole pattern here is confusing to me, we launch a separate coroutine with the same dispatcher (and an unbound scope) and then await it immediately. Why? 🤔 

I wonder if we could move the `install()` calls into `initialize()` here, before launching the coroutine, making this more explicit here? 🤔

Reply:
Thank you. Will look into this.

Reply:
Looked to be a great solution! Thank you!